templates/plugin/src/index.ts

```diff
@@ -0,0 +1,108 @@
+import type { CollectionSlug, Config } from 'payload'
+
+export type MyPluginConfig = {
+  /**
+   * List of collections to add a custom field
+   */
+  collections?: Partial<Record<CollectionSlug, true>>
+  disabled?: boolean
+}
+
+export const myPlugin =
+  (pluginOptions: MyPluginConfig) =>
+  (config: Config): Config => {
+    if (pluginOptions.disabled) {
```

Comment:
Might be worth leaving a comment here about _always_ adding any injected fields to accommodate adapters with migrations.